components/tabs/tab-autocollapse-titles.js

```diff
@@ -0,0 +1,511 @@
+      }
+
+      const allVisibleTheSame = res.visible.length === preparedElements.visible.length &&
+        res.visible.every((item, index) => item === preparedElements.visible[index]);
+      const allHiddenTheSame = res.hidden.length === preparedElements.hidden.length &&
+        res.hidden.every((item, index) => item === preparedElements.hidden[index]);
+
+      if (!allVisibleTheSame || !allHiddenTheSame || preparedElements.ready !== res.ready) {
+        setPreparedElements(res);
+      }
+    }, DEFAULT_DEBOUNCE_INTERVAL);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [children, elements.lastVisibleIndex, preparedElements, selected, selectedIndex]);
+
+  // Get list of all possibly visible elements to render in a measure container
+  const childrenToMeasure = React.useMemo(
+    () => {
+      const items = children.filter(tab => tab.props.alwaysHidden !== true);
+      return getTabTitles({items, tabIndex: -1});
+    },
+    [children]
+  );
+
+  // Initial measure for tabs and more button sizes
+  React.useEffect(() => {
+    if (measureRef.current == null) {
+      return;
+    }
+
+    const measureTask = fastdom.measure(() => {
+      const container = measureRef.current;
+      const descendants = [...container.children];
+      const moreButton = descendants.pop();
+
+      let moreButtonWidth = moreButton.offsetWidth;
+      const {
+        marginLeft: moreButtonMarginLeft,
+        marginRight: moreButtonMarginRight
+      } = getComputedStyle(moreButton);
+      moreButtonWidth += +moreButtonMarginLeft.replace('px', '') + +moreButtonMarginRight.replace('px', '');
+
+      const tabsWidth = descendants.map(node => {
+        const {marginLeft, marginRight} = getComputedStyle(node);
+        const width = node.getBoundingClientRect().width;
+        return width + +marginLeft.replace('px', '') + +marginRight.replace('px', '');
+      });
+
+      // eslint-disable-next-line no-param-reassign
+      const newSummaryWidth = tabsWidth.reduce((acc, curr) => (acc += curr), 0);
+      // eslint-disable-next-line no-param-reassign
+      const oldSummaryWidth = elements.sizes.tabs.reduce((acc, curr) => (acc += curr), 0);
+
+      if (elements.sizes.more !== moreButtonWidth || newSummaryWidth !== oldSummaryWidth) {
+        dispatch({type: ACTIONS.MEASURE_TABS, more: moreButtonWidth, tabs: tabsWidth});
+      }
+    });
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      fastdom.clear(measureTask);
+    };
+  }, [children, elements.sizes.more, elements.sizes.tabs]);
+
+  // Start observers to listen resizing and mutation
+  React.useEffect(() => {
+    let measureTask = null;
+    let resizeObserver = null;
+
+    if (measureRef.current === null) {
+      return;
+    }
+
+    resizeObserver = new ResizeObserver(entries => {
+      entries.forEach(entry => {
+        fastdom.clear(measureTask);
+        measureTask = fastdom.measure(() => adjustTabs(entry));
```

Comment:
`adjustTabs` uses `dispatch`, so it should be `mutate` not `measure`

Reply:
Should not it be like:
`fastdom.measure(() => {
       ....
       fastdom.mutate(() => {
           ....
       });
}`
       
Not simply 'fastdom.mutate'?

Reply:
just `mutate` should be enough, I don't see any measuremens there